contracts/src/components/hyperstructure.cairo

```diff
@@ -0,0 +1,20 @@
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct HyperStructure {
+    #[key]
+    entity_id: u128,
+    created_at: u64,
+    completed_at: u64,
+    resource_count: usize
+}
+
+
+
+#[derive(Component, Copy, Drop, Serde)]
+struct HyperStructureResource {
```

Comment:
do we need this? can we not use the other resource structure?

Reply:
Issue still seems to be the same as with OrderResources.

We have a Resource component where entity_id, resource_type are the keys, and we have only amount as value.

Here we nee